lib/mayaUsd/ufe/UsdUndoRenameCommand.cpp

```diff
@@ -42,18 +42,29 @@ UsdUndoRenameCommand::UsdUndoRenameCommand(const UsdSceneItem::Ptr& srcItem, con
 	: Ufe::UndoableCommand()
 {
 	const UsdPrim& prim = srcItem->prim();
-	fStage = prim.GetStage();
-	fUfeSrcItem = srcItem;
-	fUsdSrcPath = prim.GetPath();
-	// Every call to rename() (through execute(), undo() or redo()) removes
+	m_fStage = prim.GetStage();
+	m_fUfeSrcItem = srcItem;
+	m_fUsdSrcPath = prim.GetPath();
+	
+    // Every call to rename() (through execute(), undo() or redo()) removes
 	// a prim, which becomes expired.  Since USD UFE scene items contain a
 	// prim, we must recreate them after every call to rename.
-	fUsdDstPath = prim.GetParent().GetPath().AppendChild(TfToken(newName.string()));
-	fLayer = defPrimSpecLayer(prim);
-	if (!fLayer) {
+	m_fUsdDstPath = prim.GetParent().GetPath().AppendChild(TfToken(newName.string()));
+	m_fLayer = defPrimSpecLayer(prim);
+	if (!m_fLayer) {
 		std::string err = TfStringPrintf("No prim found at %s", prim.GetPath().GetString().c_str());
 		throw std::runtime_error(err.c_str());
 	}
+
+    // check if the target layer has any opinions that affects selected prim
+    if (!isTargetLayerHaveOpinion(prim)) {
+        auto possibleTargetLayer = targetLayerWithOpion(prim);
+        std::string err = TfStringPrintf("Unable to rename [%s] on current target layer, " 
+                                         "Please set [%s] as the target layer to proceed", 
+                                         prim.GetName().GetString(), 
+                                         possibleTargetLayer->GetDisplayName());
+        throw std::runtime_error(err.c_str());
+    }
```

Comment:
We check here if the target layer has any opinions that affects selected prim. If it fails, we then find the possible target layer and append it's display name to the error message. Finaly we throw a runtime_error message.